### PR TITLE
test: fix ineffectual assignments

### PR DESCRIPTION
**Pull request by idealhack**

### CODE REVIEW

File: hostdiscovery/hostdiscovery_test.go [hunk 1]
```diff
@@ -48,6 +48,9 @@ func TestAddedCallback(t *testing.T) {
 	if !added {
 		t.Fatal("Expecting an Added callback notification. But none received")
 	}
+	if removed {
+		t.Fatalf("Not expecting an Removed callback notification. But received a callback")
```
Comment (euanh) [inline hostdiscovery/hostdiscovery_test.go]: You do not need to use `Fatalf()` here - you can use `Fatal()`

File: hostdiscovery/hostdiscovery_test.go [hunk 2]
```diff
@@ -58,6 +61,9 @@ func TestRemovedCallback(t *testing.T) {
 	added := false
 	removed := false
 	hd.processCallback(update, func() {}, func(hosts []net.IP) { added = true }, func(hosts []net.IP) { removed = true })
+	if added {
+		t.Fatalf("Not expecting an Added callback notification. But received a callback")
```
Comment (euanh) [inline hostdiscovery/hostdiscovery_test.go]: `Fatalf` -> `Fatal`

File: hostdiscovery/hostdiscovery_test.go (hunk 1 above)
Comment (idealhack) [reply hostdiscovery/hostdiscovery_test.go]: Thank you, I'm on it.

File: hostdiscovery/hostdiscovery_test.go (hunk 2 above)
Comment (idealhack) [reply hostdiscovery/hostdiscovery_test.go]: fixed

**Comment by ctelfer** [comment]
Thanks for addressing this!

**Pull request merged**